go-controller/pkg/allocator/pod/pod_annotation.go

```diff
@@ -274,6 +289,16 @@ func allocatePodAnnotationWithRollback(
 	hasIPRequest := network != nil && len(network.IPRequest) > 0
 	hasStaticIPRequest := hasIPRequest && !reallocateIP
 
+	var ipamClaim *ipamclaimsapi.IPAMClaim
```

Comment:
Or maybe the correct thing to do is to ignore it, given that the ipam claim reference can be added and removed at any time. But I kinda feel we should ignore it always when the network does not allow persitent ips, regardless of whether the network config AllowsPersistentIPs is not set, or if it is the incorrect topology, or if it does not have IPAM.

Reply:
> The only doubt I am left with here is if we should error out if there is an ipam claim reference but this network does not allow persistent IPs. Versus ignoring the ipam claim reference which is what it looks like we do, or I am missing the check somewhere. Not sure if the specification says something about this or we are left to decide what to do.

IMO, we should error, log, and probably even send an event (maybe I'm going overboard ...). Ignoring will pretty much give the user something different than what was requested, which will IMO only lead to confusion.

> There is already some checks going on within FindIPAMClaim that would return error, but not the one I mention above I think. So for consistency, I would return error for the scenario above. The check could probably be done within FindIPAMClaim since you have netInfo there as well, but I find checks within FindIPAMClaim inconspicuous, I would probably surface them out of there.

I'm fine with coalescing all checks in the same place (either client or the method's implementation). I would push it to the implementation though, since it can be easily unit tested, and specified as part of the desired behavior of the ipam claims reconciler, and left as an implementation detail of the pod annotation allocator. 
I won't insist though.

Reply:
OK so after discussing this offline w/ you we've agreed on ignoring the error when the network does not allow for persistent IPs.

I've added a log for it.